lib/src/tasks/task_runner/api.dart

```diff
@@ -0,0 +1,118 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:dart_dev/util.dart' show reporter, TaskProcess;
+
+import 'package:dart_dev/src/tasks/task.dart';
+
+Future<TaskRunner> runTasks({List<String> tasksToRun}) async {
+  var taskGroup = new TaskGroup(tasksToRun);
+  await taskGroup.start();
+  taskGroup.output();
+
+  TaskRunner task = new TaskRunner(await taskGroup.checkExitCodes(),
+      taskGroup.taskGroupStdout, taskGroup.taskGroupStderr);
+  return task;
+}
+
+class TaskRunner extends Task {
+  final Future done = new Future.value();
+  final String stdout;
+  final String stderr;
+  bool successful;
+
+  TaskRunner(int exitCode, String this.stdout, String this.stderr) {
+    this.successful = exitCode == 0;
+  }
+}
+
+class SubTask {
+  /// Process to be executed
+  final String command;
+
+  /// A string consisting of all the subtask's output
+  String taskOutput = '';
+
+  /// Any output created in the subtask's stderr
+  String taskError = '';
+
+  TaskProcess taskProcess;
+
+  SubTask(this.command);
+
+  /// Starts the provided subtask and wires up a stream to allow
+  /// the TaskGroup to listen for subtask completion.
+  startProcess() {
+    var taskArguments = command.split(' ');
+    taskProcess =
+        new TaskProcess(taskArguments.first, taskArguments.sublist(1));
+    taskProcess.stdout.listen((line) {
+      this.taskOutput += '\n$line';
+    });
+    taskProcess.stderr.listen((line) {
+      this.taskOutput += '\n$line';
+      this.taskError += '\n$line';
+    });
+  }
+}
+
+class TaskGroup {
+  /// List of the individual subtasks executed
+  List<String> subTaskCommands = [];
+
+  /// List of the subtasks making up the TaskGroup
+  List<SubTask> subTasks = [];
+
+  /// TaskGroup stdout
+  String taskGroupStdout = '';
+
+  /// TaskGroup stderr
+  String taskGroupStderr = '';
+
+  TaskGroup(List<String> this.subTaskCommands) {
```

Comment:
I don't think you need the type here if you are referring to the class member `subTaskCommands`.